internal/helm/getter/transport.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package getter
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+)
+
+type TransportPool struct {
+}
```

Comment:
Can you add a doc block here describing what it can be used for, and why we put it into place over the Helm default? Plus maybe some instructions about how using it renders the TLS getter options useless (and needs to be provided using the `tlsConfig` argument).

Reply:
@hiddeco I tried to add as much context as possible whilst not being too verbose. PTAL